ph5/core/ph5api.py

```diff
@@ -1454,111 +1454,59 @@ def get_availability(self, das, sample_rate, component,
-                            float(entry['sample_rate_i']) /
-                            float(entry['sample_rate_multiplier_i']))
-                else:
-                    previous['length'] = 0
-                    previous['sample_rate'] = 0
-
-                previous['end'] = previous['start'] + previous['length']
-        if gaps > 0:
-            start_ = (
-                    float(new_das_t[-1]['time/epoch_l']) +
-                    float(new_das_t[-1]['time/micro_seconds_i']) /
-                    1000000)
-            if new_das_t[-1]['sample_rate_i'] > 0:
-                length_ = (
-                        float(new_das_t[-1]['sample_count_i']) /
-                        float(new_das_t[-1]['sample_rate_i']) /
-                        float(new_das_t[-1]['sample_rate_multiplier_i']))
-                sample_rate_ = (
-                        float(new_das_t[-1]['sample_rate_i']) /
-                        float(new_das_t[-1]['sample_rate_multiplier_i']))
-            else:
-                length_ = 0
-                sample_rate_ = 0
-            end_ = start_ + length_
-            times.append((sample_rate_, start_, end_))
-
-        if gaps == 0:
-            start_time = (
-                    float(new_das_t[0]['time/epoch_l']) +
-                    float(new_das_t[0]['time/micro_seconds_i']) /
-                    1000000)
-            if new_das_t[0]['sample_rate_i'] > 0:
-                sample_rate_ = (
-                        float(new_das_t[0]['sample_rate_i']) /
-                        float(new_das_t[0]['sample_rate_multiplier_i']))
-            else:
-                sample_rate_ = 0
-
-            start_ = (
-                    float(new_das_t[-1]['time/epoch_l']) +
-                    float(new_das_t[-1]['time/micro_seconds_i']) /
-                    1000000)
-            if new_das_t[-1]['sample_rate_i'] > 0:
-                length_ = (
-                        float(new_das_t[-1]['sample_count_i']) /
-                        float(new_das_t[-1]['sample_rate_i']) /
-                        float(new_das_t[-1]['sample_rate_multiplier_i']))
-                sample_rate_ = (
-                        float(new_das_t[-1]['sample_rate_i']) /
-                        float(new_das_t[-1]['sample_rate_multiplier_i']))
+            if not previous:
+                previous['start'] = cur_time
+                previous['length'] = cur_len
+                previous['sample_rate'] = cur_sr
+                previous['end'] = cur_time + cur_len
             else:
-                length_ = 0
-                sample_rate_ = 0
-            end_ = start_ + length_
-
-            times.append((sample_rate_, start_time, end_))
+                if cur_time == previous['start'] and \
+                        cur_len == previous['length'] and \
+                        cur_sr == previous['sample_rate']:
+                    # duplicate entry - skip
+                    continue
```

Comment:
Does this happen? often? Just curious.

Reply:
I'm not sure how often this happens, but I did come across cases in older experiments where there were duplicate records at the very beginning of the Das_t dataset.  If I remember correctly, experiment 14-021 was one such case.